perun-core/src/main/java/cz/metacentrum/perun/core/entry/FacilitiesManagerEntry.java

```diff
@@ -131,12 +131,15 @@ public Facility getFacilityByName(PerunSession sess, String name) throws Interna
 		if (AuthzResolver.isAuthorized(sess, Role.PERUNADMIN)) {
 			return getFacilitiesManagerBl().getRichFacilities(sess);
 		} else if (AuthzResolver.isAuthorized(sess, Role.FACILITYADMIN)) {
-			// Cast complementary object to Facility
-			List<Facility> facilities = new ArrayList<>();
-			for (PerunBean facility: AuthzResolver.getComplementaryObjectsForRole(sess, Role.FACILITYADMIN, Facility.class)) {
-				facilities.add((Facility) facility);
-			}
-			//Now I create list of richFacilities from facilities
+			List<Facility> facilities = getFacilitiesManagerBl().getFacilities(sess);
+			facilities.removeIf(facility -> {
+				try {
+					return !AuthzResolver.isAuthorized(sess, Role.FACILITYADMIN, facility);
+				} catch (InternalErrorException e) {
+					// if we can't determine authorization prevent returning it
+					return true;
+				}
+			});
```

Comment:
I am afraid this is not effective for situations where there are lot of facilities and I am admin of just small amount of them. In situation like 100K facilities exists and I am admin only for 2 of them this is very ineffective call. (even if the situation is very unlikely)

Reply:
For now we have max. hundreds of facilities, so it should be quicker to get data at once than loading 5-10 facilities by id. But I added comment/todo explaining the situation. We should load proper facilities directly, but right now that would be inconsistent with contract of our authz. We keep authorization in session, which is not reflecting authz changes before its refresh. As we agreed, I'm merging this.